KIPs/kip-82.md

```diff
@@ -0,0 +1,376 @@
+
+As of September 21st, 2022, of the 300 million KLAY that is minted annually, only 100 million KLAY goes to GC, and 20% and 80% of the distribution is from this 100 million KLAY. The renewal separates GC reward into two groups: basic reward and staked reward. The basic reward is compensating for node operation by providing 20% of GC reward and 100% of gas fee resources from transactions. The staked reward provides only the rest of 80% of the minting resource based on the total stake amount.
+
+The relative ratio of 20% and 80% is a tentative value to be used in the Cypress mainnet. However, this ratio is defined as a governance paramter GC members can change the ratio through the governance process.
+
+![gc reward allocation](../assets/kip-82/gc_reward_allocation.png)
+
+After the Magma hard fork, Klaytn has been burning the first half of the gas fee based on [the KIP-71 proposal](https://kips.klaytn.foundation/KIPs/kip-71) while the second half is distributed. The second half is distributed into GC reward, Klaytn Improvement Reserve (KIR) and Klaytn Growth Fund (KGF).
+
+The renewal will include a change of gas fee as follows.
+
+In addition to changing the reward structure, this renewal will include burning the gas fee that is provided as GC’s basic reward. The KGF and KIR portion will be excluded from the gas fee.
+If the amount of gas fee that is provided as a basic reward does not exceed a certain level, all of them will be burnt. The threshold amount is 20% of minting provided as a basic reward. If the gas fee exceeds this amount, 20% will be incinerated and the additional amount will be distributed to GC.
+
+To reduce the KLAY supply amount due to inflation, the gas fee will be burnt.
+
+### Klaytn node update
+
+The Klaytn node will be updated according to the new reward policy. The CN portion is further split into basic reward and staked reward. The staked reward is then distributed among CNs proportional to their staking shares. Their shares are calculated proportionally to their stake amounts except for the minimum staking amount.
+
+Two new governance parameters fine control the new reward distribution algorithm. The default values below will be used in the Cypress mainnet. The parameters can be updated by the existing governance mechanism.
+
+| Name                | Type   | Meaning                                              | Default Value |
+|---------------------|--------|------------------------------------------------------|---------------|
+| `reward.usekip82`   | bool   | Whether the new algorithm is used or not             | false         |
+| `reward.kip82ratio` | string | The relative ratio between basic and staking rewards | "20/80"       |
+
+During integer arithmetics, minuscule remaining amounts may be emitted as by-products. Such remaining amounts will be added to the proposer or KGF portion. Below pseudocode illustrates the new reward distribution algorithm. Note: `//` is round down integer division.
```

Comment:
> Such remaining amounts will be added to the proposer or KGF portion.

@blukat29 what is the condition of the two distribution? when the remaining amount is added to the proposer? and when to KGF?

Reply:
proposer: remainder of staking distribution
kgf: remainder of distribution of proposer, kir, kgf

Reply:
```suggestion
During integer arithmetics, minuscule remaining amounts may be emitted as by-products. Such remaining amounts will be added to the proposer or KGF portion. The proposer gets the remainder of staking distribution while the KGF gets the remainder of the distribution among the block proposer, KIR, and KGF. Below pseudocode illustrates the new reward distribution algorithm. Note: `//` is round down integer division.
```